docs2/site/docs/getting-started/errors.md

```diff
@@ -1,78 +1,149 @@
+
+Here is a sample of a typical unhandled exception delegate which logs the error to a database.  It also returns the log id along with the error message:
+
+```csharp
+var executer = new DocumentExecuter();
+var result = executer.ExecuteAsync(options => {
+
+    ...
+
+    options.UnhandledExecutionDelegate = context =>
+    {
+        try
+        {
+            using var db = new MyDatabaseContext();
+            var errorLog = new ErrorLog {
+                DateStamp = DateTime.UtcNow,
+                Message = context.Exception.Message,
+                Details = context.Exception.ToString()
+            };
+            db.ErrorLogs.Add(errorLog);
+            db.SaveChanges();
+            context.Exception.Data["errorLogId"] = errorLog.Id;
+        }
+        catch { }
+    };
+});
 ```
 
-Also `ExecutionOptions.UnhandledExceptionDelegate` allows you to override, hide,
-modify or just log the unhandled exception from your resolver before wrap it into
-`ExecutionError`. This can be useful for hiding error messages that reveal server
-implementation details.
+You can also override the serialized exception by setting `context.Exception`, or simply replace the message by setting `context.ErrorMessage` such as in this example:
 
 ```csharp
-var executor = new DocumentExecutor();
-ExecutionResult result = await executor.ExecuteAsync(_ =>
+options.UnhandledExecutionDelegate = ctx =>
 {
-  _.Query = "...";
-  _.UnhandledExceptionDelegate = context => context.ErrorMessage = "Error Message";
-});
+    if (ctx.Exception is SqlException)
+        ctx.ErrorMessage = "A database error has occurred.";
+};
 ```
 
-Set `ExecutionOptions.ThrowOnUnhandledException` to `true` in cases where the thrown exception needs to propagate to an enclosing or global handler.
+## Error Serialization
 
-Note that `UnhandledExceptionDelegate` will not be invoked in the case `ThrowOnUnhandledException` is `true`.
+After the `DocumentExecuter` has returned a `ExecutionResult` containing the data and/or errors, typically you will pass this object to an implementation of `IDocumentWriter` to convert the object tree into json. The GraphQL spec allows for four properties to be returned with each error: `message`, `locations`, `path`, and `extensions`. The `IDocumentWriter` implementations provided for the `Newtonsoft.Json` and `System.Text.Json` libraries allow you to control the serialization of `ExecutionError`s into the resulting json data by providing an `IErrorInfoProvider` to the constructor of the document writer. The `ErrorInfoProvider` class contains 5 properties to control serialization behavior:
```

Comment:
```suggestion
After the `DocumentExecuter` has returned a `ExecutionResult` containing the data and/or errors, typically you will pass this object to an implementation of `IDocumentWriter` to convert the object tree into json. The GraphQL spec allows for four properties to be returned with each error: `message`, `locations`, `path`, and `extensions`. The `IDocumentWriter` implementations provided for the `Newtonsoft.Json` and `System.Text.Json` libraries allow you to control the serialization of `ExecutionError`s into the resulting json data by providing an `IErrorInfoProvider` to the constructor of the document writer. The `ErrorInfoProvider` class (default implementation of `IErrorInfoProvider`) contains 5 properties to control serialization behavior:
```

Reply:
Github does not show changes in a line if it is very long. Consider break up very long lines.

Reply:
Good idea.